libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -429,14 +426,69 @@ public boolean checkBirthDateCorrelationInCredentials(String userId)
+    // validate this
+    // requirement.
+    private List<IdentityClaim> getIdentityClaimsForBirthDateCorrelation(
+            List<VcStoreItem> vcStoreItems) throws HttpResponseExceptionWithErrorBody {
+        List<IdentityClaim> identityClaims = new ArrayList<>();
+        for (VcStoreItem item : vcStoreItems) {
+            IdentityClaim identityClaim =
+                    getIdentityClaim(item.getCredential(), item.getCredentialIssuer(), true);
+            if (isBirthDateEmpty(identityClaim.getBirthDate())) {
+                // excluded BAV and Address CRI
+                if (EXCLUDE_CRIS_TYPES_FOR_DOB_CORRELATION.contains(item.getCredentialIssuer())) {
+                    continue;
+                }
+                addLogMessage(item, "Birthdate property is missing from VC");
+            }
+            identityClaims.add(identityClaim);
+        }
+        return identityClaims;
+    }
+
+    private boolean isBirthDateEmpty(List<BirthDate> birtDates) {
+        return CollectionUtils.isEmpty(birtDates)
+                || birtDates.stream().map(BirthDate::getValue).allMatch(StringUtils::isEmpty);
+    }
+
     public boolean checkNameAndFamilyNameCorrelationInCredentials(String userId)
             throws HttpResponseExceptionWithErrorBody, CredentialParseException {
         final List<VcStoreItem> successfulVCStoreItems =
                 getSuccessfulVCStoreItems(getVcStoreItems(userId));
-        List<IdentityClaim> identityClaims = getIdentityClaims(successfulVCStoreItems);
+        List<IdentityClaim> identityClaims =
+                getIdentityClaimsForNameCorrelation(successfulVCStoreItems);
         return checkNamesForCorrelation(getFullNamesFromCredentials(identityClaims));
     }
 
+    private List<IdentityClaim> getIdentityClaimsForNameCorrelation(List<VcStoreItem> vcStoreItems)
+            throws HttpResponseExceptionWithErrorBody {
+        List<IdentityClaim> identityClaims = new ArrayList<>();
+        for (VcStoreItem item : vcStoreItems) {
+            IdentityClaim identityClaim =
+                    getIdentityClaim(item.getCredential(), item.getCredentialIssuer(), true);
+            if (isNamesEmpty(identityClaim.getName())) {
+                // excluded Address CRI. Because Address CRI does not contain a name,
+                if (ADDRESS_CRI.equals(item.getCredentialIssuer())) {
+                    continue;
+                }
+                addLogMessage(item, "Name property is missing from VC");
+            }
+            identityClaims.add(identityClaim);
+        }
+        return identityClaims;
+    }
+
+    private boolean isNamesEmpty(List<Name> names) {
+        return CollectionUtils.isEmpty(names)
+                || names.stream()
+                        .flatMap(name -> name.getNameParts().stream())
+                        .map(NameParts::getValue)
+                        .allMatch(StringUtils::isEmpty);
```

Comment:
(ditto)

Reply:
If names list comes as null., a NullPointerException will be thrown. I think, we should keep it.

Reply:
When is it `null`, rather than empty list?

Reply:
I don't know in which scenario it might come as null, but wouldn't it be better to safely it, just in case?

Reply:
Not necessarily - if we don't expect this to return null, then it could be concealing an error.

However, if you think it might be null in certain cases that aren't a problem, then this is fine :)

(the 'billion-dollar mistake'...)